From 432ee78984ca94560ce3145748ba2ef6069dc825 Mon Sep 17 00:00:00 2001
From: Camber Huang <camber@aosc.io>
Date: Sun, 17 Sep 2023 06:30:50 +0000
Subject: [PATCH] Replace Pandoc with ronn-ng

As pandoc is unavailable to architectures other than amd64 in AOSC OS,
generation of manpages with upstream-shipped repository is not possible. This
patch replaces pandoc with architecture-independent ronn-ng, along with minor
tweaks to markdown source files.

This patch will also replace previous patch which just simply disables the
generation of manpage.

This patch should be dropped when haskell compiler is available on other
architectures.

Signed-off-by: Camber Huang <camber@aosc.io>
---
 dev-bin/make-man-pages.pl | 13 ++++++-------
 doc/GeoIP.conf.md         | 12 ++++++------
 doc/geoipupdate.md        | 18 +++++++++---------
 3 files changed, 21 insertions(+), 22 deletions(-)

diff --git a/dev-bin/make-man-pages.pl b/dev-bin/make-man-pages.pl
index 82acb07..f191e5c 100755
--- a/dev-bin/make-man-pages.pl
+++ b/dev-bin/make-man-pages.pl
@@ -33,13 +33,12 @@ sub _make_man {
     close $fh or die $!;
 
     system(
-        'pandoc',
-        '-s',
-        '-f', 'markdown',
-        '-t', 'man',
-        $tmp,
-        '-o', $output,
-    ) == 0 or die 'pandoc failed';
+        'ronn',
+        '--roff',
+        '--name=' . $name,
+        '--section=' . $section,
+        $input
+    ) == 0 or die 'ronn(ronn-ng) failed';
 
     return;
 }
diff --git a/doc/GeoIP.conf.md b/doc/GeoIP.conf.md
index 6ba9ca5..46454ef 100644
--- a/doc/GeoIP.conf.md
+++ b/doc/GeoIP.conf.md
@@ -2,18 +2,18 @@
 
 GeoIP.conf - Configuration file for geoipupdate
 
-# SYNOPSIS
+## SYNOPSIS
 
 This file allows you to configure your `geoipupdate` program to
 download GeoIP2 and GeoLite2 databases.
 
-# DESCRIPTION
+## DESCRIPTION
 
 The file consists of one setting per line. Lines starting with `#`
 are comments and will not be processed. All setting keywords are case
 sensitive.
 
-## Required settings:
+### Required settings:
 
 `AccountID`
 
@@ -35,7 +35,7 @@ sensitive.
     at run time by the `GEOIPUPDATE_EDITION_IDS` environment variable. Note:
     this was formerly called `ProductIds`.
 
-## Optional settings:
+### Optional settings:
 
 `DatabaseDirectory`
 
@@ -92,7 +92,7 @@ sensitive.
     overridden at run time by the `GEOIPUPDATE_PARALLELISM` environment
     variable or the `--parallelism` command line argument.
 
-## Deprecated settings:
+### Deprecated settings:
 
 The following are deprecated and will be ignored if present:
 
@@ -102,6 +102,6 @@ The following are deprecated and will be ignored if present:
 
 `SkipHostnameVerification`
 
-# SEE ALSO
+## SEE ALSO
 
 `geoipupdate`(1)
diff --git a/doc/geoipupdate.md b/doc/geoipupdate.md
index e39b88d..98d9e5e 100644
--- a/doc/geoipupdate.md
+++ b/doc/geoipupdate.md
@@ -2,11 +2,11 @@
 
 geoipupdate - GeoIP2 and GeoLite2 Update Program
 
-# SYNOPSIS
+## SYNOPSIS
 
 **geoipupdate** [-Vvh] [-f *CONFIG_FILE*] [-d *TARGET_DIRECTORY*]
 
-# DESCRIPTION
+## DESCRIPTION
 
 `geoipupdate` automatically updates GeoIP2 and GeoLite2 databases. The
 program connects to the MaxMind GeoIP Update server to check for new
@@ -16,7 +16,7 @@ install it.
 If you are using a firewall, you must have the DNS and HTTPS ports
 open.
 
-# OPTIONS
+## OPTIONS
 
 `-d`, `--database-directory`
 
@@ -56,11 +56,11 @@ open.
 
 :   Output download/update results in JSON format.
 
-# EXIT STATUS
+## EXIT STATUS
 
 `geoipupdate` returns 0 on success and 1 on error.
 
-# NOTES
+## NOTES
 
 Typically you should run `geoipupdate` weekly. On most Unix-like systems,
 this can be achieved by using cron. Below is a sample crontab file that
@@ -79,11 +79,11 @@ To use with a proxy server, update your `GeoIP.conf` file as specified in
 the `GeoIP.conf` man page. Alternatively, set the `GEOIPUPDATE_PROXY` or
 `http_proxy` environment variable.
 
-# BUGS
+## BUGS
 
 Report bugs to [support@maxmind.com](mailto:support@maxmind.com).
 
-# AUTHORS
+## AUTHORS
 
 Written by William Storey.
 
@@ -92,11 +92,11 @@ This software is Copyright (c) 2018-2023 by MaxMind, Inc.
 This is free software, licensed under the Apache License, Version 2.0 or
 the MIT License, at your option.
 
-# MORE INFORMATION
+## MORE INFORMATION
 
 Visit [our website](https://www.maxmind.com/en/geoip2-services-and-databases)
 to learn more about the GeoIP2 databases or to sign up for a subscription.
 
-# SEE ALSO
+## SEE ALSO
 
 `GeoIP.conf`(5)
-- 
2.39.1

